apps/server/src/modules/task-card/controller/api-test/task-card.api.spec.ts

```diff
@@ -90,164 +178,179 @@ describe('Task-Card Controller (api)', () => {
-			expect(responseTaskCard.task.taskCardId).toEqual(responseTaskCard.id);
-			expect(responseTaskCard.task.status.isDraft).toEqual(false);
+			currentUser = mapUserToCurrentUser(user);
+
+			const taskCardParams = {};
+
+			await request(app.getHttpServer())
+				.post(`/cards/task/`)
+				.set('Accept', 'application/json')
+				.send(taskCardParams)
+				.expect(400);
 		});
-		it('DELETE should remove task-card, its card-elements and associated task', async () => {
-			const user = setupUser([Permission.TASK_CARD_EDIT, Permission.HOMEWORK_EDIT]);
+	});
+	describe('[GET] /cards/task/:id', () => {
+		it('should return existing task-card', async () => {
+			const user = setupUser([Permission.TASK_CARD_VIEW, Permission.HOMEWORK_VIEW]);
+			const title = 'title test';
 			// for some reason taskCard factory messes up the creator of task, so it needs to be separated
-			const task = taskFactory.build({ creator: user });
-			const taskCard = taskCardFactory.build({
-				creator: user,
-				cardElements: [richTextCardElementFactory.buildWithId()],
-				task,
-			});
+			const task = taskFactory.build({ name: title, creator: user });
+			const taskCard = taskCardFactory.buildWithId({ creator: user, task });
 
 			await em.persistAndFlush([user, task, taskCard]);
 			em.clear();
 
-			const cardElementsIds = taskCard.getCardElements().map((element) => element.id);
-			const foundCardElementsInitial = await em.findAndCount(CardElement, { id: { $in: cardElementsIds } });
-			expect(foundCardElementsInitial[1]).toEqual(1);
-
 			currentUser = mapUserToCurrentUser(user);
 
-			await request(app.getHttpServer())
-				.delete(`/cards/task/${taskCard.id}`)
+			const response = await request(app.getHttpServer())
+				.get(`/cards/task/${taskCard.id}`)
 				.set('Accept', 'application/json')
 				.expect(200);
 
-			const foundCardElements = await em.findAndCount(CardElement, { id: { $in: cardElementsIds } });
-			expect(foundCardElements[1]).toEqual(0);
+			const responseTaskCard = response.body as TaskCardResponse;
 
-			const foundTask = await em.findOne(Task, { id: taskCard.task.id });
-			expect(foundTask).toEqual(null);
+			expect(responseTaskCard.id).toEqual(taskCard.id);
+		});
+		it('should throw if feature not enabled', async () => {
+			await cleanupCollections(em);
+			Configuration.set('FEATURE_TASK_CARD_ENABLED', false);
+			const user = setupUser([]);
+			const taskCard = taskCardFactory.buildWithId({ creator: user });
 
-			const foundTaskCard = await em.findOne(TaskCard, { id: taskCard.id });
-			expect(foundTaskCard).toEqual(null);
+			await em.persistAndFlush([user, taskCard]);
+			em.clear();
+
+			currentUser = mapUserToCurrentUser(user);
+
+			await request(app.getHttpServer())
+				.get(`/cards/task/${taskCard.id}`)
+				.set('Accept', 'application/json')
+				.expect(500);
 		});
-		it('PATCH should update the task card', async () => {
+	});
+	describe('[PATCH] /cards/task/:id', () => {
```

Comment:
what about the validation errors in the PATCH section?

Reply:
@MartinSchuhmacher this pull request is not about validation and testing. I just used it to restructure the file (what was not the best idea) and didn't add, remove or modify any tests. I would prefer to add validation errors in a FollowUp.

Reply:
https://ticketsystem.dbildungscloud.de/browse/BC-3840